File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int len = 0;
        int n = nums.size();
        for(int i = 0; i<n; i++){
            int j = i+1; 
            int incr = 1;
            while(j < n && nums[j-1] < nums[j]){
                incr++;
                j++;
            }
            j = i+1;
            int decr = 1;
            while(j < n && nums[j-1] > nums[j]){
                decr++;
                j++;
            }
            len = max(max(incr,decr), len);
        }
        return len;
    }
};

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int len = 1;
        int incr = 1, decr = 1;
        int n = nums.length;
        for(int i = 1; i<n; i++){
            // check for strictly increasing..
            if(nums[i-1] < nums[i]){
                incr++;
                decr = 1;
                len = Math.max(incr,len);
            }else if(nums[i-1] > nums[i]){// strictly decreasing
                decr++;
                incr = 1;
                len = Math.max(decr,len);
            }else{ // both are equal
                incr = 1;
                decr = 1;
            }
        }
        return len;
    }
}